tests/run-make/doctests-keep-binaries/rmake.rs

```diff
@@ -0,0 +1,71 @@
+// Check that valid binaries are persisted by running them, regardless of whether the
+// --run or --no-run option is used. 
+
+use run_make_support::{run, rustc, rustdoc, tmp_dir};
+use std::fs::{create_dir, remove_dir_all};
+use std::path::Path;
+
+fn setup_test_env<F: FnOnce(&Path, &Path)>(callback: F) {
+    let out_dir = tmp_dir().join("doctests");
+    create_dir(&out_dir).expect("failed to create doctests folder");
+    rustc()
+        .input("t.rs")
+        .crate_type("rlib")
+        .run();
+    callback(&out_dir, &tmp_dir().join("libt.rlib"));
+    remove_dir_all(out_dir);
+}
+
+fn check_generated_binaries() {
+    run("doctests/t_rs_2_0/rust_out");
+    run("doctests/t_rs_8_0/rust_out");
+}
+
+fn main() {
+    setup_test_env(|out_dir, extern_path| {
+        rustdoc()
+            .input("t.rs")
+            .arg("-Zunstable-options")
+            .arg("--test")
+            .arg("--persist-doctests")
+            .arg(out_dir)
+            .arg("--extern")
+            .arg(format!("t={}", extern_path.display()))
```

Comment:
Do we want any helper methods on `Rustdoc` for some of the stable flags? e.g.

```rs
rustdoc()
    // ...
    .extern("t", extern_path) // or whatever
```

Reply:
Hum... Considering there are only 3 doctests in the `run-make` testsuite, is it really worth it?